metadata.md

```diff
@@ -0,0 +1,116 @@
+# Metadata Schema - Version 0.1.0
+
+In order to ensure interoperability between implementations of the Arrow in-memory format
+and associated storage formats such as Parquet and Feather, it is necessary to
+include additional top-level metadata.
+
+This information includes:
+
+-   spatial properties of each geometry column
+-   name of the primary geometry column
+
+## Schema stability
+
+This is the first major draft of this metadata schema. It is intended to help
+test implementation details and gather feedback from the community. As such,
+this version does not make any stability promises that will be upheld in future
+versions. We encourage you to offer feedback on this schema and implement
+tests using it. We do not encourage you to use this version of the schema
+or associated implementations in production settings until future versions of
+this schema.
+
+## Metadata structure
+
+Metadata is to be stored as a JSON-encoded UTF-8 string under the "geo" key within the
+top-level metadata object of the in-memory Arrow schema or file format.
+
+For clarity, the following shows the unencoded JSON structure:
+
+```python
+"geo": {
+        "columns": {...},  # REQUIRED: see below
+        "creator": {
+            "library": "<library name>",  # REQUIRED
+            "version": "<library version>",  # REQUIRED
+        },
+        "primary_column": "<primary geometry column name>",  # REQUIRED
+        "version": "1.0.0"  # REQUIRED
+    }
+```
+
+Each geometry column in the dataset must be included in the `columns` field above
+with the following content, keyed by the column name:
+
+```python
+"<column_name>": {
+    "bounds": [<xmin>, <ymin>, (zmin), <xmax>, <ymax>, (zmax)],  # OPTIONAL
```

Comment:
Maybe we should use "bbox" instead of "bounds"? ("bbox" is also the term used in eg GeoJSON spec)